src/client/app/components/HeaderComponent.tsx

```diff
@@ -7,61 +7,106 @@ import { Link } from 'react-router';
 import { Button } from 'react-bootstrap';
 import LogoComponent from './LogoComponent';
 import UIModalComponent from './UIModalComponent';
-import getToken from '../utils/getToken';
+import { deleteToken, hasToken} from '../utils/token';
 
 interface HeaderProps {
-	renderLoginButton?: boolean;
-	renderGroupsButton?: boolean;
-	renderOptionsButton?: boolean;
-	title?: string;
+	title: string;
 }
 
 /**
  * React component that controls the header strip at the top of all pages
  * @param props The props passed down by the parent component
  * @return JSX to create the header strip
  */
-export default function HeaderComponent(props: HeaderProps) {
-	const title = props.title ? props.title : 'Open Energy Dashboard';
-	const titleStyle = {
-		display: 'inline-block'
-	};
-	const divRightStyle = {
-		float: 'right',
-		marginTop: '5px',
-		marginRight: '20px'
-	};
-	const loginLinkStyle = {
-		// Displays the login button link only if the user is not logged in or is explicitly told to render
-		display: (getToken() || props.renderLoginButton === false) ? 'none' : 'inline',
-		paddingLeft: '5px'
-	};
-	const adminLinkStyle = {
-		// Displays the admin button link only if the user is logged in (auth token exists)
-		display: getToken() ? 'inline' : 'none',
-		paddingLeft: '5px'
-	};
-	const groupsLinkStyle = {
-		// Displays the groups button link only if the user is logged in (auth token exists) or explicitly told to render
-		display: (getToken() && props.renderGroupsButton) ? 'inline' : 'none',
-		paddingLeft: '5px'
-	};
-	return (
-		<div className='container-fluid'>
-			<div className='col-xs-4'>
-				<Link to='/'><LogoComponent url='./app/images/logo.png' /></Link>
-			</div>
-			<div className='col-xs-4 text-center'>
-				<h1 style={titleStyle}>{title}</h1>
-			</div>
-			<div style={divRightStyle}>
-				<div className='visible-sm visible-xs'>
-					{(props.renderOptionsButton) ? <UIModalComponent /> : null}
+export default class HeaderComponent extends React.Component<HeaderProps, {}> {
```

Comment:
I very much like this refactored HeaderComponent, thanks for this. Much less convoluted.